iroha-balancer-client/src/main/kotlin/soramitsu/irohautils/balancer/client/service/IrohaBalancerClientService.kt

```diff
@@ -0,0 +1,112 @@
+package soramitsu.irohautils.balancer.client.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.rabbitmq.client.BuiltinExchangeType
+import com.rabbitmq.client.Connection
+import com.rabbitmq.client.ConnectionFactory
+import com.rabbitmq.client.impl.DefaultExceptionHandler
+import iroha.protocol.TransactionOuterClass
+import jp.co.soramitsu.iroha.java.Utils
+import mu.KLogging
+import soramitsu.irohautils.balancer.client.config.RMQConfig
+import java.io.Closeable
+
+open class IrohaBalancerClientService @JvmOverloads constructor(
+        val rmqConfig: RMQConfig,
+        private val onRmqFail: () -> Unit = {
+            logger.error { "RMQ failure. Exit." }
+            throw ClientErrorException("RMQ has failed. Please, check settings and the error.")
+        }
+) : Closeable {
+
+    private val factory = ConnectionFactory()
+
+    private val objectMapper = ObjectMapper()
+
+    private val connection by lazy {
+        if (rmqConfig.username == null || rmqConfig.password == null) {
+            logger.error { "RMQ credentials are not provided" }
+            throw ClientErrorException("RMQ credentials are not provided. Please, check your settings.")
```

Comment:
Is it valid to throw ClientErrorException in this place? How the client can correct it?